examples/tv-app/android/java/MyUserPrompter-JNI.cpp

```diff
@@ -182,17 +233,78 @@ bool JNIMyUserPrompter::DisplaysPasscodeAndQRCode()
 void JNIMyUserPrompter::PromptWithCommissionerPasscode(uint16_t vendorId, uint16_t productId, const char * commissioneeName,
                                                        uint32_t passcode, uint16_t pairingHint, const char * pairingInstruction)
 {
-    // TODO
-    ChipLogError(Zcl, "JNIMyUserPrompter::PromptWithCommissionerPasscode Needs Implementation");
+    ChipLogError(Zcl, "JNIMyUserPrompter::PromptWithCommissionerPasscode");
+
+    DeviceLayer::StackUnlock unlock;
+    CHIP_ERROR err = CHIP_NO_ERROR;
+    JNIEnv * env   = JniReferences::GetInstance().GetEnvForCurrentThread();
+    std::string stringCommissioneeName(commissioneeName);
+    std::string stringPairingInstruction(pairingInstruction);
+
+    VerifyOrExit(mJNIMyUserPrompterObject.HasValidObjectRef(), err = CHIP_ERROR_INCORRECT_STATE);
+    VerifyOrExit(mPromptForCommissionOKPermissionMethod != nullptr, err = CHIP_ERROR_INCORRECT_STATE);
+    VerifyOrExit(env != nullptr, err = CHIP_JNI_ERROR_NO_ENV);
+
+    {
+        UtfString jniCommissioneeName(env, stringCommissioneeName.data());
+        UtfString jniPairingInstruction(env, stringPairingInstruction.data());
+        env->ExceptionClear();
+        env->CallVoidMethod(mJNIMyUserPrompterObject.ObjectRef(), mPromptWithCommissionerPasscodeMethod,
+                            static_cast<jint>(vendorId), static_cast<jint>(productId), jniCommissioneeName.jniValue(),
+                            static_cast<jlong>(passcode), static_cast<jint>(pairingHint), jniPairingInstruction.jniValue());
+        if (env->ExceptionCheck())
+        {
+            ChipLogError(DeviceLayer, "Java exception in PromptWithCommissionerPasscode");
+            env->ExceptionDescribe();
+            env->ExceptionClear();
+            err = CHIP_ERROR_INCORRECT_STATE;
+            goto exit;
+        }
+    }
+
+exit:
+    if (err != CHIP_NO_ERROR)
+    {
+        ChipLogError(Zcl, "PromptWithCommissionerPasscode error: %s", err.AsString());
+    }
 }
 
 /**
  *   Called to alert the user that commissioning has begun."
  */
 void JNIMyUserPrompter::PromptCommissioningStarted(uint16_t vendorId, uint16_t productId, const char * commissioneeName)
 {
-    // TODO
-    ChipLogError(Zcl, "JNIMyUserPrompter::PromptCommissioningStarted Needs Implementation");
+    ChipLogError(Zcl, "JNIMyUserPrompter::PromptCommissioningStarted");
+
+    DeviceLayer::StackUnlock unlock;
+    CHIP_ERROR err = CHIP_NO_ERROR;
+    JNIEnv * env   = JniReferences::GetInstance().GetEnvForCurrentThread();
+    std::string stringCommissioneeName(commissioneeName);
+
+    VerifyOrExit(mJNIMyUserPrompterObject.HasValidObjectRef(), err = CHIP_ERROR_INCORRECT_STATE);
+    VerifyOrExit(mPromptForCommissionOKPermissionMethod != nullptr, err = CHIP_ERROR_INCORRECT_STATE);
+    VerifyOrExit(env != nullptr, err = CHIP_JNI_ERROR_NO_ENV);
+
+    {
+        UtfString jniCommissioneeName(env, stringCommissioneeName.data());
```

Comment:
same as the above, use env->NewStringUTF, thanks

Reply:
addressed